modules/platforms/cpp/thin-client/src/impl/transactions/transactions_proxy.cpp

```
@@ -0,0 +1,80 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "ignite/impl/thin/transactions/transactions_proxy.h"
+#include "impl/transactions/transactions_impl.h"
+
+using namespace ignite::impl::thin;
+using namespace transactions;
+
+namespace
+{
+    using namespace ignite::common::concurrent;
+
+    TransactionsImpl& GetTxsImpl(SharedPointer<void>& ptr)
+    {
+        return *reinterpret_cast<TransactionsImpl*>(ptr.Get());
+    }
+
+    TransactionImpl& GetTxImpl(SharedPointer<void>& ptr)
+    {
+        return *reinterpret_cast<TransactionImpl*>(ptr.Get());
+    }
+}
+
+namespace ignite
+{
+    namespace impl
+    {
+        namespace thin
+        {
+            namespace transactions
+            {
+                TransactionProxy TransactionsProxy::txStart(
+                        TransactionConcurrency::Type concurrency,
+                        TransactionIsolation::Type isolation,
+                        int64_t timeout,
+                        int32_t txSize,
+                        const char *lbl)
+                {
+                    return TransactionProxy(GetTxsImpl(impl).TxStart(concurrency, isolation, timeout, txSize, lbl));
+                }
+
+                void TransactionProxy::commit()
+                {
+                    GetTxImpl(impl).Commit();
+                }
+
+                void TransactionProxy::rollback()
+                {
+                    GetTxImpl(impl).Rollback();
+                }
+
+                void TransactionProxy::close()
+                {
+                    try
+                    {
+                        GetTxImpl(impl).Close();
+                    }
+                    catch (...)
```

Empty catch block? Do you think that it is good idea?

---

yep, close can`t throw any exceptions i hope.

---

I understand why. But i  suppose, that std::set<int32_t> under locks + pointer to transaction in thread_local is better.

---

I'm not sure about this solution to be honest. This can lead to really hard-to-debug bugs.